src/component/GamePlay/GamePlay.jsx

```diff
@@ -1,93 +1,142 @@
+
 import "./GamePlay.css";
 import React, { useEffect, useState } from "react";
 import NavBar from "../nav_bar/nav_bar.component";
-import { useLocation } from "react-router-dom";
+import { useSearchParams } from "react-router-dom";
+
+
+function normalizeAnimalData(animalData) {
+  if (!animalData) return null;
+
+  if (animalData.data) {
+    return {
+      id: animalData.data.id,
+      attributes: animalData.data.attributes,
+    };
+  }
+
+  return {
+    id: animalData.id,
+    attributes: animalData.attributes,
+  };
+}
 
```

Comment:
Great way to handle and normalize the various data types we are receiving.  Great stuff.